File: RainForecast4/src/com/kve/rainforecast4/location/WeatherLocation.java
package com.kve.rainforecast4.location;

import java.util.Arrays;
import java.util.List;

import android.text.TextUtils;

public class WeatherLocation {

	private String name;
	private String latitudeStr;
	private String longitudeStr;
	private Double latitude;
	private Double longitude;

	//	No info
	public WeatherLocation()
	{
		initEmptyValues();
	}
	
	//	All info
	public WeatherLocation(String aName, String aLatitude, String aLongitude)
	{
		initWithValues(aName, aLatitude, aLongitude);
	}
	
	//	Name and combined Lat and Long
	public WeatherLocation(String aName, String aLatLong)
	{
		String[] latLongParts = TextUtils.split(aLatLong, " : ");
		if (latLongParts.length == 2)
		{
			initWithValues(aName, latLongParts[0], latLongParts[1]);
		}
		else
		{
			initEmptyValues();
		}
	}
	
	//	All info combined as per toString()
	public WeatherLocation(String locationInfoString)
	{
		List <String> locationParts = Arrays.asList(TextUtils.split(locationInfoString, ":"));
		if (locationParts.size() != 3)
		{
			initEmptyValues();
		}
		else
		{
			initWithValues(locationParts.get(0),
					locationParts.get(1).replace("latitude=", ""),
					locationParts.get(2).replace("longitude=", ""));
		}
	}

	
	private void initEmptyValues()
	{
		setName("");
		setLatitude("");
		setLongitude("");
	}

	private void initWithValues(String aName, String aLatitude, String aLongitude)
	{
		setName(aName);
		setLatitude(aLatitude);
		setLongitude(aLongitude);
	}

	
	public String getName() {
		return name;
	}
	private void setName(String name) {
		this.name = name;
	}
	public Double getLatitude() {
		return latitude;
	}
	private void setLatitude(String latitude) {
		this.latitudeStr = latitude;
		try {
			this.latitude = Double.parseDouble(latitudeStr);
		} catch (NumberFormatException e) {
			// TODO Auto-generated catch block
			this.latitude = 0.0;
		}
	}
	public Double getLongitude() {
		return longitude;
	}
	private void setLongitude(String longitude) {
		this.longitudeStr = longitude;
		try {
			this.longitude = Double.parseDouble(longitudeStr);
		} catch (NumberFormatException e) {
			// TODO Auto-generated catch block
			this.longitude = 0.0;
		}
	}

	public String toString()
	{
		String weatherLocString = name + ":latitude="+ latitude + ":longitude=" + longitude;
		return weatherLocString;
	}
}

File: SnowForecast/app/src/main/java/com/sners/snowforecast/location/WeatherLocation.kt
package com.sners.snowforecast.location

import android.text.TextUtils
import java.lang.NumberFormatException
import java.util.*

/**
 * A class representing a location, for storing in the menu
 * I'm not messing with this class until I have set up my menus
 */
class WeatherLocation {
    var name: String? = null
        private set
    private var latitudeStr: String? = null
    private var longitudeStr: String? = null
    var latitude: Double? = null
        private set
    var longitude: Double? = null
        private set

    //	No info
    constructor() {
        initEmptyValues()
    }

    //	All info
    constructor(aName: String, aLatitude: String, aLongitude: String) {
        initWithValues(aName, aLatitude, aLongitude)
    }

    //	Name and combined Lat and Long
    constructor(aName: String, aLatLong: String?) {
        val latLongParts = TextUtils.split(aLatLong, " : ")
        if (latLongParts.size == 2) {
            initWithValues(aName, latLongParts[0], latLongParts[1])
        } else {
            initEmptyValues()
        }
    }

    //	All info combined as per toString()
    constructor(locationInfoString: String?) {
        val locationParts = Arrays.asList(*TextUtils.split(locationInfoString, ":"))
        if (locationParts.size != 3) {
            initEmptyValues()
        } else {
            initWithValues(
                locationParts[0],
                locationParts[1].replace("latitude=", ""),
                locationParts[2].replace("longitude=", "")
            )
        }
    }

    private fun initEmptyValues() {
        setName("")
        setLatitude("")
        setLongitude("")
    }

    private fun initWithValues(aName: String, aLatitude: String, aLongitude: String) {
        setName(aName)
        setLatitude(aLatitude)
        setLongitude(aLongitude)
    }

    private fun setName(name: String) {
        this.name = name
    }

    private fun setLatitude(latitude: String) {
        latitudeStr = latitude
        try {
            this.latitude = latitudeStr!!.toDouble()
        } catch (e: NumberFormatException) {
            // TODO Auto-generated catch block
            this.latitude = 0.0
        }
    }

    private fun setLongitude(longitude: String) {
        longitudeStr = longitude
        try {
            this.longitude = longitudeStr!!.toDouble()
        } catch (e: NumberFormatException) {
            // TODO Auto-generated catch block
            this.longitude = 0.0
        }
    }

    override fun toString(): String {
        return "$name:latitude=$latitude:longitude=$longitude"
    }
}